connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -421,6 +424,32 @@ message StreamingQueryManagerCommandResult {
   }
 }
 
+// The protocol for client-side StreamingQueryListener.
+// This command will only be set when either the first listener is added to the client, or the last
+// listener is removed from the client.
+// The add_listener_bus_listener command will only be set true in the first case.
+// The remove_listener_bus_listener command will only be set true in the second case.
+message StreamingQueryListenerBusCommand {
+  oneof command {
+    bool add_listener_bus_listener = 1;
+    bool remove_listener_bus_listener = 2;
+  }
+}
+
+// The protocol for the returned events in the long-running response channel.
+// The event_type for QueryProgressEvent is 1;
+// for QueryTerminatedEvent is 2; for QueryIdleEvent is 3;
+// There is no QueryStartedEvent defined here,
```

Comment:
Should we formalize this somehow?

Reply:
Yes we can use similar mechanism as PythonEvalType
https://github.com/databricks/runtime/blob/95f1afae584a89d20ecf2e411f8b3a267d354e46/python/pyspark/rdd.py#L141

Reply:
oh I see what you meant, createrd a enum in proto file